shell/platform/linux/public/flutter_linux/fl_dart_project.h

```diff
@@ -0,0 +1,27 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_DART_PROJECT_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_DART_PROJECT_H_
+
+#include <glib-object.h>
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+G_BEGIN_DECLS
+
+G_DECLARE_FINAL_TYPE(FlDartProject, fl_dart_project, FL, DART_PROJECT, GObject)
+
+FlDartProject* fl_dart_project_new(const gchar* assets_path,
```

Comment:
As a follow-up PR, it would be good to move this to align with https://github.com/flutter/engine/pull/17210. At the level of this API, we should enforce a folder structure for all the artifacts we expect (both to simplify the API, and to make it more future-proof), and since we're writing the tooling that people will be expected to use to generate that folder they won't even need to know the details.

Reply:
I've made this change in another branch, but can fold it into this PR if you like.